### PR TITLE
Experimental PE file internal dependency walker

### PR DESCRIPTION
**Pull request by deajan**
### What does this PR do?

Enable internal PE dependency walker on pyd / dll files for Windows environment.
Works with ```--experimental=useInternalDependencyWalker``` and optionally ```--experimental=recursiveInternalDependencies```

On some packages that have multiple dependencies in different site-packages directories, it's also useful to add ```--experimental=recurseIntoSitePackagesDependencyWalker``` in order to check all site-packages for dependent .pyd/.dll files.

### Why was it initiated? Any relevant Issues?

Slow dependency walks with depends.exe, also includes a lot of not always used DLLs.
Also fixes missing pythoncom37.dll dependency when using pywin32
Using recurseIntoSitePackagesDependencyWalker may fix other missing DLLs (PyQT5 for example).

### PR Checklist
- [X] Correct base branch selected? `develop` for new fetures and bug fixes too. If it's
      part of a hotfix, it will be moved to ``master`` during it.
- [X] All tests still pass. Check the developer manual about ``Running the Tests``. There
      are Travis tests that cover the most important things however, and you are
      welcome to rely on those, but they might not cover enough.
- [X] Ideally new features or fixed regressions ought to be covered via new tests.
Not applicable
- [ ] Ideally new or changed features have documentation updates.

### CODE REVIEW

**Comment by deajan** [comment]
I'm willing to write unit tests, but I have no clue where to begin. Ideas ?

**Comment by kayhayen** [comment]
Looks pretty good to me already, I hope you can try the tests, and I am OK with you getting it to work for your target platform.

There is already a list of to be ignored DLLs, and I wonder if this should not be shared, or split into another file, I think one for Linux got added recently too, but lets keep these distinct for now.

Good job!

**Comment by kayhayen** [comment]
And just a silly question, do you mean, `do merge` or `do not merge` in the title.

File: nuitka/freezer/Standalone.py [hunk 1]
```diff
@@ -995,20 +1181,30 @@ def detectBinaryDLLs(is_main_executable, source_dir, original_filename,
         of used DLLs is retrieved.
     """
 
-
     if Utils.getOS() in ("Linux", "NetBSD", "FreeBSD"):
         return _detectBinaryPathDLLsLinuxBSD(
             dll_filename = original_filename
         )
     elif Utils.getOS() == "Windows":
         with TimerReport("Running depends.exe for %s took %%.2f seconds" % binary_filename):
```
[the remark this answers is not in the document]
Comment (deajan) [reply nuitka/freezer/Standalone.py]: Done in latest commit

**Comment by deajan** [comment]
I meant 'no ready TO merge' indeed :)
Thinking of the tests, I won't be able to write travis tests since the target platform is Windows and cross compilation isn't a thing right now. I improved the code so at least pylint passes.

As for the DLL lists, I've added two DLLs to the list, so it might change the current non experimental behavior sharing the list.

I have asked @JorjMcKie if he could make some tests and he came back some QT usecases where the dependency walker can't find the complete path of the DLLs, hence failing with an AssertionError.
The problem is that I can't just walk through the whole site-pacakges directory in hope to find the right DLLs. I'll try to see what I can do.
Once this is resolved, I'll ask you considering to merge :)

**Comment by deajan** [comment]
Thing is I don't want to get os architecture but Python architecture, since running Python 32 on Windows 64 is possible, and I can compile 32 bit executables on Windows 64.

**Comment by deajan** [comment]
I have made a lot of smaller improvements, with the help of @JorjMcKie, and my PR should be ready for merge I think.
Hoping this may become the de-facto implementation in the future.

Grüsse aus Sud-Frankreich :)

**Comment by deajan** [comment]
Just forgot to talk about docs, is there anywhere I should write documentation for the experimental features ?

**Comment by kayhayen** [comment]
@deajan there is not a lot of user doc for this, you are welcome to write up a small outline how this works for the develop manual, adding a section, but I wouldn't require it.

I want to merge this next as I am waiting with #217 where `ldd.exe` can be used, but touches the same files, so I was holding off on that.

If you don't mind, I would rename the experimental flags to something shorter, say `use_pefile` or so, just so it's easier to type. 

And it's really cool to see so much collaboration going on outside of me, @JorjMcKie, glad to see these contributions. I will then try and run comparative tests between pefile and dependency walker code, and report back. 

Comparing the produced folders might totally make sense to find bugs. One of the advantages of the experimental approach is to be able to do that easily. Then we need to decide if we want to go prime time with it, or go experimental first for a few releases, asking people to try it.

**Comment by deajan** [comment]
I am willing to write a small usage doc, can you point me to the right files to do so ?

As for ```ldd.exe```, I actually used it too before switching to pefile (also tried objdump, both cygwin versions), but ldd can't get imports on ```.pyd``` files whereas pefile can.
So the pefile dependency walker actually does check every pyd file too for missing dependencies.

Of course, I will rename to ```use_pefile, use_pefile_recurse, and use_pefile_fullrecurse``` if you're happy with this.

I am impatient to see this going "live", and having a user base testing it :)
So far, my tests included compiling a dozen executables of which windows services, tkinter, qt and REST api programs.

**Comment by kayhayen** [comment]
I think `README.rst` should be where you document the usability of the standalone mode. I am going to check out in comparison testing the old against the new pefile based one, and will create issues as I go @deajan 

I think it would be nice if you just posted a small description in the issue, and then I can reference it. My goal is to make your code the baseline, and hopefully to make it have only one decision for the user that works by default. Multi-homed packages like Qt so far got it extra paths via the plugin for it, so maybe it's not needed.

I am also hopeful, this `pefile` approach allows for proper dependency management, where we notice when files needed change, and update things properly. For `depends.exe` this was never quite done, but I have to look at it. But I do not consider it urgent.

The `pefile` is maintained unlike `depends.exe` where e.g. I think issues with unicode paths are much harder to deal with.

Thanks a lot for this contribution, I am very hopeful this is a huge step forward for Windows and working standalone support.

**Comment by deajan** [comment]
Thank you for your kind words :) I'll write something up for the docs after rebasing my develop branch.

**Pull request merged**